Sources/HttpPipeline/NIO.swift

```diff
@@ -107,6 +107,10 @@ private final class Handler: ChannelInboundHandler {
       }
     }
   }
+
+  func errorCaught(ctx: ChannelHandlerContext, error: Error) {
+    _ = ctx.close()
```

Comment:
if you use `ctx.close(promise: nil)` you can save one allocation for when you don't care about the result anyway (I realise my example code didn't do that ;) )

Reply:
Good to know, thanks! We've only started to dig deeper into NIO from our current lightweight wrapper. We've got a branch going that redefines middleware as `(Conn<I, A>) -> EventLoopFuture<Conn<J, B>>` everywhere instead of our current stack built on `IO`: https://github.com/pointfreeco/swift-web/compare/nio-conn

Haven't been able to fully flesh it out yet, though!

Reply:
wow, nice. That sounds awesome!